verify_urls.go

```diff
@@ -21,18 +22,31 @@ func main() {
     fileContent := string(readmeFile)
     urlElementRegex := regexp.MustCompile(`(?m)\[.+?]\(((http|https)://.+?)\)`)
 
-    httpClient := http.Client{Timeout: 20 * time.Second}
+    httpClient := http.Client{
+        Timeout: 20 * time.Second,
+        Transport: &http.Transport{
+            TLSClientConfig: &tls.Config{},
+        },
+    }
 
     var brokenUrls []string
     for _, urlElement := range urlElementRegex.FindAllStringSubmatch(fileContent, -1) {
         var url = urlElement[1]
 
         fmt.Printf("Checking %s: ", url)
 
-        resp, err := httpClient.Get(url)
+        req, err := http.NewRequest("GET", url, nil)
+        req.Header.Add("User-Agent", "URL status code verification for the Flyeralarm onboarding resources; https://github.com/flyeralarm/onboarding")
+        resp, err := httpClient.Do(req)
+
+        errormessage := err
+        if errormessage == nil {
+            errormessage = errors.New(http.StatusText(resp.StatusCode))
+        }
+
```

Comment:
Please add `defer resp.Body.Close()` when there was no error.
See https://stackoverflow.com/questions/33238518/what-could-happen-if-i-dont-close-response-body for reference

Reply:
Deferring would only make sense if the function would return at this point. But since the function is always executed till the end, deferring wouldn't make sense - or am I missing something?